api/server/sdk/volume_migrate.go

```diff
@@ -0,0 +1,143 @@
+/*
+Package sdk is the gRPC implementation of the SDK gRPC server
+Copyright 2018 Portworx
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package sdk
+
+import (
+	"context"
+
+	"github.com/libopenstorage/openstorage/api"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// Start a volume migration
+func (s *VolumeServer) Start(
+	ctx context.Context,
+	req *api.SdkCloudMigrateStartRequest,
+) (*api.SdkCloudMigrateStartResponse, error) {
+
+	if len(req.GetName()) == 0 {
```

Comment:
All redundant checks. Should be done by the implementation not here.

Reply:
See comment on 4xx and 5xx messages. We will try to keep these to a minimum

Reply:
Again, overreach from the SDK. In case the name isn't required to be passed in for the Portworx implementation, so this check is actually incorrect.

Reply:
Name is required for idempotency, as All SDK calls are idempotent. The SDK and CSI are both idempotent. The Golang implementations may not be.

Reply:
A taskId is not required. It will be generated if not supplied. The golang implementation is idempotent. Just requiring a name here doesn't make anything idempotent. The implementation needs to guarantee that.

Reply:
talked offline, @disrani-px will check in #696 checks. He will point out any that do not need checks.